examples/vpc_peering_common_dual_stack/main.tf

```diff
@@ -0,0 +1,275 @@
+  content = templatefile("templates/bootstrap_common.tmpl",
+    {
+      trust_gcp_router_ip   = each.value.bootstrap_template_map.trust_gcp_router_ip
+      private_network_cidr  = each.value.bootstrap_template_map.private_network_cidr
+      untrust_gcp_router_ip = each.value.bootstrap_template_map.untrust_gcp_router_ip
+      trust_loopback_ip     = each.value.bootstrap_template_map.trust_loopback_ip
+      untrust_loopback_ip   = each.value.bootstrap_template_map.untrust_loopback_ip
+      trust_loopback_ipv6   = each.value.bootstrap_template_map.trust_loopback_ipv6
+      untrust_loopback_ipv6 = each.value.bootstrap_template_map.untrust_loopback_ipv6
+    }
+  )
+}
+
+resource "local_sensitive_file" "init_cfg" {
+
+  for_each = { for k, v in var.vmseries : k => v
+    if can(v.bootstrap_template_map)
+  }
+
+  filename = "files/${each.key}/config/init-cfg.txt"
+  content = templatefile(
+    "templates/init-cfg.tmpl",
+    { bootstrap_options = merge(var.vmseries_common.bootstrap_options, each.value.bootstrap_options) }
+  )
+}
+
+module "bootstrap" {
+  source = "../../modules/bootstrap"
+
+  for_each = var.bootstrap_buckets
+
+  folders = keys(var.vmseries)
+
+  name_prefix     = "${var.name_prefix}${each.value.bucket_name_prefix}"
+  service_account = module.iam_service_account[each.value.service_account_key].email
+  location        = each.value.location
+  files = merge(
+    { for k, v in var.vmseries : "files/${k}/config/bootstrap.xml" => "${k}/config/bootstrap.xml" if can(v.bootstrap_template_map) },
+    { for k, v in var.vmseries : "files/${k}/config/init-cfg.txt" => "${k}/config/init-cfg.txt" if can(v.bootstrap_template_map) },
+  )
+}
+
+module "vpc" {
+  source = "../../modules/vpc"
+
+  for_each = var.networks
+
+  project_id                      = var.project
+  name                            = "${var.name_prefix}${each.value.vpc_name}"
+  create_network                  = each.value.create_network
+  delete_default_routes_on_create = each.value.delete_default_routes_on_create
+  mtu                             = each.value.mtu
+  routing_mode                    = each.value.routing_mode
+  enable_ula_internal_ipv6        = try(each.value.enable_ula_internal_ipv6, false)
+  internal_ipv6_range             = try(each.value.internal_ipv6_range, null)
+  subnetworks = { for k, v in each.value.subnetworks : k => merge(v, {
+    name = "${var.name_prefix}${v.name}"
+    })
+  }
+  firewall_rules = try({ for k, v in each.value.firewall_rules : k => merge(v, {
+    name = "${var.name_prefix}${v.name}"
+    })
+  }, {})
+}
+
+resource "google_compute_route" "this" {
+
+  for_each = var.routes
+
+  name         = "${var.name_prefix}${each.value.name}"
+  dest_range   = each.value.destination_range
+  network      = module.vpc[each.value.vpc_network_key].network.self_link
+  next_hop_ilb = module.lb_internal[each.value.lb_internal_key].forwarding_rule
+  priority     = 100
+}
+
+resource "null_resource" "policy_routes" {
```

Comment:
What is the purpose of these policy based routes ? I can see that it's pointing the default ipv6 route to the LB - I'm guessing there is no resource right now to add ipv6 routes to point to the LBs ?

My only fear with this is that the interpreter is bash and no everyone has MACOS/Linux - and Windows VMs will fail this provision of a null resource.

Reply:
I described limitations and workarounds in the README [here](https://github.com/PaloAltoNetworks/terraform-google-swfw-modules/tree/feature/ipv6-examples-vpc-peering-common-dual-stack/examples/vpc_peering_common_dual_stack#ipv6-routing).
Windows users should have WSL, but let me check if we can do it better.